### PR TITLE
Implemented test matrices: pascal and toeplitz

### PR DESCRIPTION
**Pull request by Lengjunyi**
For pascal, only the symmetric form of the matrix is implemented.
For toeplitz, the implemented functionality is as with Matlab, except for consideration for complex numbers

### CODE REVIEW

**Comment by lidangzzz** [comment]
@Gaoooooo Could you help with this pull request? Thanks

File: pascal.hhs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    pascalSquare[row][0] = 1;
+  }
+
+  // Use the definition of pascal matrix
+  for (let row = 1; row < n; row++) {
+    for (let col = 1; col < n; col++) {
+      pascalSquare[row][col] = pascalSquare[row-1][col] + pascalSquare[row][col-1];
+    }
+  }
+  return new Mat(pascalSquare);
```
Comment (Gaoooooo) [inline pascal.hhs:40]:
```suggestion
return pascalSquare;
```

File: pascal_test.hhs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        if (inv(A).val.some(c => c.some(x => !isInteger(x)))) {
+            throw new Error('Unit test failed for pascal');
+        }
+    }
+
+    function isInteger(x) {
+        return Number.isInteger(Math.round(x * 100000) / 100000);
+    }
+    
+    print('pascal test pass');
```
Comment (Gaoooooo) [inline pascal_test.hhs:24]:
```suggestion
//print('pascal test pass');
```

**Review by Gaoooooo — COMMENTED** [review]
Firstly, I apologize for being quite late for this pull request. We were bogged down with other related work. This looks great, only thing I changed manually is to return a JS array instead of a Mat (newer design approach) and made the print statement a comment, as we don't want test cases actually printing if succeeding. Thank you very much.

File: toeplitz.hhs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    // Fill other rows
+    for (let i = 1; i < m; i++) {
+      for (let j = 1; j < n; j++) {
+        toeplitzMatrix[i][j] = toeplitzMatrix[i-1][j-1]
+      }
+    }
+
+    return new Mat(toeplitzMatrix);
```
Comment (Gaoooooo) [inline toeplitz.hhs:110]:
```suggestion
return toeplitzMatrix;
```

File: toeplitz.hhs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    // Fill other rows
+    for (let i = 1; i < n; i++) {
+      for (let j = 0; j < n; j++) {
+        toeplitzSquare[i][j] = r[Math.abs(i - j)]
+      }
+    }
+
+    return new Mat(toeplitzSquare);
```
Comment (Gaoooooo) [inline toeplitz.hhs:66]:
```suggestion
return toeplitzSquare;
```

**Review by Gaoooooo — COMMENTED** [review]
Similar to Pascal, I changed it so it returns JS arrays. 

Also, should it be considered to write a check for symmetry in toeplitz_r()?

**Comment by Gaoooooo** [comment]
Also, I added toeplitz, pascal to all.hhs, and pascal_test to test.hhs

**Pull request merged**